src/rcx/src/extmeasure.cpp

```diff
@@ -249,23 +249,19 @@ ext2dBox* extMeasure::addNew2dBox(dbNet* net,
 {
   ext2dBox* bb = _2dBoxPool->alloc();
 
+  std::array<int, 2> bb_ll;
+  std::array<int, 2> bb_ur;
   dbShape s;
   if ((net != NULL) && _extMain->getFirstShape(net, s)) {
-    bb->_ll[0] = s.xMin();
-    bb->_ll[1] = s.yMin();
-    bb->_ur[0] = s.xMax();
-    bb->_ur[1] = s.yMax();
+    bb_ll = {s.xMin(), s.yMin()};
+    bb_ur = {s.xMax(), s.yMax()};
   } else {
-    bb->_ll[0] = ll[0];
-    bb->_ll[1] = ll[1];
-    bb->_ur[0] = ur[0];
-    bb->_ur[1] = ur[1];
+    bb_ll = {ll[0], ll[1]};
+    bb_ur = {ur[0], ur[1]};
   }
 
-  bb->_met = m;
-  bb->_dir = d;
-  bb->_id = id;
-  bb->_map = 0;
+  assert(d == 0 || d == 1);
```

Comment:
Should this be moved to a logger error instead of an assert?

Reply:
Good review question -- is there OpenROAD developer guidance on when to prefer logging error (and having a corresponding exception thrown) vs assertion / why? Right now it basically seems like an invariant that the direction value was either 0 or 1. Since we're using `<cassert>` we don't see if there's an invariant violation in opt/release mode with the way it's done here.

Reply:
assert will crash the whole program, whereas error will only exit the current command. In most CI user configurations the result is the same. However, in interactive mode this may be less desirable.

Reply:
It does imply that invalid values may already exist and be floating around in in-memory objects, so the question is really whether "keep on trucking" is better in those circumstances. (Which is a value judgement.)

If so, it sounds a reasonable policy would be something like, "if there's a logger already accessible in your scope, prefer to assert using the logger error instead of a C assert". Will switch to that.

Reply:
Logger error is preferred.

Reply:
I'm also not sure how these error IDs are allocated, I grepped and eyeballed what the max one seemed to be and added one. Since I imagine this error is not very user-relevant I'm not sure if there's a good canonical way to flag that it's an internal error in the error string? LMK if so.

Reply:
Ethan pointed out: https://openroad.readthedocs.io/en/latest/contrib/Logger.html#coding That resolves this question. Thanks!